### DISCUSSION TITLE
I am new at Python can someone help me? (for loop and functions)

### DISCUSSION THREAD

**Discussion by planckmechanics** [community/community#21469] [Other Feature Feedback, Questions, & Ideas]
<p>Guys, I am new at Python and ı am learning it myself. I am started to this: Prime numbers to 1 to “n”. n is input. Can you tell me where is the my fault.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/d/3d3ad6b5d22f8be61e625c816b7421d9e7b42ca8.png" data-download-href="/uploads/short-url/8JFbB2J0eIyw3v74EcdzbuIpDLO.png?dl=1" title="Adsız" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/73117214/181064882-956cfd23-6f93-4508-9c99-26b91ff5f280.png" alt="Adsız" data-base62-sha1="8JFbB2J0eIyw3v74EcdzbuIpDLO" width="481" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/d/3d3ad6b5d22f8be61e625c816b7421d9e7b42ca8_2_481x500.png, https://aws1.discourse-cdn.com/github/original/3X/3/d/3d3ad6b5d22f8be61e625c816b7421d9e7b42ca8.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/3/d/3d3ad6b5d22f8be61e625c816b7421d9e7b42ca8.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/d/3d3ad6b5d22f8be61e625c816b7421d9e7b42ca8_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Adsız</span><span class="informations">686×712 39.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I am using PyCharm.</p>
<p>Thanks :slight_smile:</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Your problem is that <code>liste</code> is declared globally. This means that after checking 2 it will contain this: <code>[1, 2]</code>. Then when 3 gets checked it’ll look like this after the modulo loop completes: <code>[1, 2, 1, 3]</code>, which makes the check fail.</p>
<p>Declare <code>liste</code> at the beginning of the <code>asal_sayı</code> function to solve the problem.</p>

**Comment by planckmechanics** [comment]
<p>Thank you so much :slight_smile:</p>
